libs/src/services/creatorNode/CreatorNodeSelection.js

```diff
@@ -185,16 +185,22 @@ class CreatorNodeSelection extends ServiceSelection {
 
     // If not enough secondaries returned from services, select from backups
     remainingSecondaries = this.numberOfNodes - 1 - secondaries.length
-    while (remainingSecondaries-- > 0) {
+    while (remainingSecondaries > 0) {
       const backup = this.selectFromBackups()
-      if (backup && backup !== primary) secondaries.push(backup)
+      if (backup && backup !== primary) {
+        secondaries.push(backup)
+        remainingSecondaries--
+      }
```

Comment:
won't this while infinite loop if you never enter the if block? lets say you never find a valid backup, you would just keep retrying?